### PR TITLE
Slice triangles by plane, adapting implementation from Trimesh

### PR DESCRIPTION
**Pull request by paulmelnikow**

### CODE REVIEW

File: polliwog/plane/_trimesh_intersections.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                np.zeros((0, 3), dtype=np.uint64),
+            )
+            return empty
+
+        # TODO: Adapt the optimized `unique_bincount()` helper from trimesh, which
+        # unfortunately does not work with unsigned ints.
+        # https://github.com/numpy/numpy/issues/17760
+        unique, inverse = np.unique(new_faces.reshape(-1), return_inverse=True)
+        # `np.unique()` returns signed ints.
+        inverse = inverse.astype(np.uint64)
```
Comment (paulmelnikow) [inline polliwog/plane/_trimesh_intersections.py:142]: Note modification ^^

File: polliwog/plane/_trimesh_intersections.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+    # If no faces to cut, the surface is not in contact with this plane.
+    # Thus, return a mesh with only the inside faces
+    if len(cut_faces_quad) + len(cut_faces_tri) == 0:
+
+        if len(new_faces) == 0:
+            # if no new faces at all return empty arrays
+            empty = (
+                np.zeros((0, 3), dtype=np.float64),
+                np.zeros((0, 3), dtype=np.uint64),
```
Comment (paulmelnikow) [inline polliwog/plane/_trimesh_intersections.py:133]: ^^ modification

File: polliwog/plane/_trimesh_intersections.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        tri_int_inds = np.where(cut_signs_tri == -1)[1]
+        tri_int_verts = cut_faces_tri[range(num_tris), tri_int_inds].reshape(
+            num_tris, 1
+        )
+
+        # Fill out new triangles with the intersection points as vertices
+        new_tri_faces = np.append(
+            tri_int_verts,
+            np.arange(
+                len(new_vertices), len(new_vertices) + 2 * num_tris, dtype=np.uint64
```
Comment (paulmelnikow) [inline polliwog/plane/_trimesh_intersections.py:215]: ^^ modification

File: polliwog/plane/_trimesh_intersections.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        # Append new vertices and new faces
+        new_vertices = np.append(new_vertices, new_tri_vertices, axis=0)
+        new_faces = np.append(new_faces, new_tri_faces, axis=0)
+
+    # TODO: Adapt the optimized `unique_bincount()` helper from trimesh, which
+    # unfortunately does not work with unsigned ints.
+    # https://github.com/numpy/numpy/issues/17760
+    unique, inverse = np.unique(new_faces.reshape(-1), return_inverse=True)
+    # `np.unique()` returns signed ints.
+    inverse = inverse.astype(np.uint64)
```
Comment (paulmelnikow) [inline polliwog/plane/_trimesh_intersections.py:237]: ^^ modification

**Comment by paulmelnikow** [comment]
@jbeard4

**Review by jbeard4 — APPROVED** [review]

**Pull request merged**